### ISSUE TITLE
🗑️ Deprecation promotion of `toNegativeInt` function in `ResultContext` to hidden level

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#474]
## 📝 Description

We would like to promote the deprecation of the `toNegativeInt` function in `ResultContext` to a hidden level.

## 🔗 Dependencies

This issue is blocked by the following ones:
- [x] #473

## ✅ Checklist

- [ ] Promote the deprecation of this declaration, update the public [API binaries] and the [unreleased changelog].
- [ ] Close this issue as completed and update tracking ones if present.

[api binaries]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#checking-the-api-binaries
[unreleased changelog]: https://github.com/kotools/types/blob/main/CHANGELOG.md#unreleased

**Comment by LVMVRQUXL** [comment]
Another issue for this task will be created when ready.
